src/lib.rs

```diff
@@ -138,22 +170,25 @@ fn makecircle(
     num_vertices: usize,
     projected: bool,
 ) -> Polygon<f64> {
-    let mut circle_points = Vec::new();
-    if projected {
-        for i in 0..num_vertices {
-            let angle: f64 = 2.0 * std::f64::consts::PI / (num_vertices as f64) * (i as f64);
-            let x = centerpoint.x() + radius * angle.cos();
-            let y = centerpoint.y() + radius * angle.sin();
-            circle_points.push(Point::new(x, y));
-        }
+    let circle_points: Vec<Point<f64>> = if projected {
+        (0..num_vertices)
```

Comment:
@Robinlovelace, just a quick explanation of how [collect](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect) works.

Using a for loop to build up a vector looks like this:

```
let mut points = Vec::new();
for idx in 0..num_vertices {
  // Do stuff to idx in order to produce a point
  let point = ...;
  points.push(point);
}
// Now `points: Vec<Point<f64>>` is filled out
```

Using iterators with collect, it becomes

```
let points = (0..num_vertices)
  .iter()
  .map(|idx| {
    // Do the same transformation of idx into a Point
    ...
   })
   .collect();
```

That's the gist of it. Similar to R, you can also sprinkle in `filter()` calls to skip some of the values. And earlier, there's a use of `flat_map` that does warrant a little more explanation